src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java

```diff
@@ -168,12 +168,15 @@ private CaseDetails getCaseDetails(final String caseReference) {
             .orElseThrow(() -> new ResourceNotFoundException(format("Case with reference %s could not be found", caseReference)));
     }
 
-    private CaseDetails saveCaseDetails(final CaseDetails caseDetails,
+    private CaseDetails saveCaseDetails(CaseDetails caseDetailsBefore, final CaseDetails caseDetails,
                                         final CaseEvent eventTrigger,
                                         final Optional<String> state) {
         if (!state.isPresent() && !equalsIgnoreCase(CaseState.ANY, eventTrigger.getPostState())) {
             caseDetails.setState(eventTrigger.getPostState());
         }
+        if (!caseDetails.getState().equalsIgnoreCase(caseDetailsBefore.getState())) {
+            caseDetails.setLastStateModifiedDate(LocalDateTime.now(ZoneOffset.UTC));
```

Comment:
just an improvement suggestion. Feel free to ignore. 
Usually when working with dates, to make testability simpler, it's recommended to have a DateService with methods like now() and others which can then be mocked in Unit tests. We could introduce that

Reply:
Yes, I agree. I'll see if i can introduce "Clock" class.

Reply:
done